src/configuration.cc

```diff
@@ -132,10 +133,16 @@ int MetadataAgentConfiguration::ParseArguments(int ac, char** av) {
 }
 
 void MetadataAgentConfiguration::ParseConfigFile(const std::string& filename) {
-  std::lock_guard<std::mutex> lock(mutex_);
   if (filename.empty()) return;
+  std::ifstream ifs;
+  ifs.open (filename, std::ifstream::in);
```

Comment:
How about combining this with construction, i.e.:
```cpp
std::ifstream input(filename);
ParseConfiguration(input);
```
? BTW, `in` is the default mode.

Reply:
Much better, thanks.